test/current/BuildDotFhirDotOrgClient.test.ts

```diff
@@ -0,0 +1,459 @@
+                        url: 'http://hl7.org/fhir/us/core/ImplementationGuide/hl7.fhir.us.core.r4-4.0.0',
+                        name: 'USCoreR4',
+                        'package-id': 'hl7.fhir.us.core.r4',
+                        'ig-ver': '4.0.0',
+                        date: 'Mon, 20 Jan, 2020 19:36:43 +0000',
+                        errs: 1496,
+                        warnings: 36,
+                        hints: 228,
+                        version: '4.0.0',
+                        tool: '4.1.0 (3)',
+                        repo: 'HL7/US-Core-R4/branches/main/qa.json'
+                      },
+                      {
+                        url: 'http://hl7.org/fhir/sushi-test-no-download/ImplementationGuide/sushi-test-no-download-0.1.0',
+                        name: 'sushi-test-no-download',
+                        'package-id': 'sushi-test-no-download',
+                        'ig-ver': '0.1.0',
+                        repo: 'sushi/sushi-test-no-download/branches/master/qa.json'
+                      },
+                      {
+                        url: 'http://hl7.org/fhir/sushi-test/ImplementationGuide/sushi-test-0.1.0',
+                        name: 'sushi-test',
+                        'package-id': 'sushi-test',
+                        'ig-ver': '0.1.0',
+                        repo: 'sushi/sushi-test/branches/master/qa.json'
+                      },
+                      {
+                        url: 'http://hl7.org/fhir/sushi-test/ImplementationGuide/sushi-test-0.1.0',
+                        name: 'sushi-test',
+                        'package-id': 'sushi-test',
+                        'ig-ver': '0.2.0',
+                        repo: 'sushi/sushi-test/branches/testbranch/qa.json'
+                      },
+                      {
+                        url: 'http://hl7.org/fhir/sushi-test/ImplementationGuide/sushi-test-0.1.0',
+                        name: 'sushi-test',
+                        'package-id': 'sushi-test',
+                        'ig-ver': '0.2.0',
+                        repo: 'sushi/sushi-test/branches/oldbranch/qa.json'
+                      },
+                      {
+                        url: 'http://hl7.org/fhir/sushi-no-main/ImplementationGuide/sushi-no-main-0.1.0',
+                        name: 'sushi-no-main',
+                        'package-id': 'sushi-no-main',
+                        'ig-ver': '0.1.0',
+                        repo: 'sushi/sushi-no-main/branches/feature/qa.json'
+                      },
+                    ]
+                  };
+                
+                } else if (
+                  uri === 'https://build.fhir.org/ig/HL7/US-Core-R4/branches/main/package.tgz' 
+                ) {
+                  return {
+                      status: 200,
+                      data: Readable.from(['zipfile'])
+                  };
+                
+                
+                // is the not found case needed with new impl?
```

Comment:
Sure -- let's add a test for when something is not found, which I think would more likely be a response with a HTTP 404 error. The bit in the fixture below would more likely represent an unexpected error during download, which we can also go ahead and test for too.